src/nreplClient.ts

```diff
@@ -42,31 +42,39 @@ interface nREPLCloseMessage {
     op: string;
 }
 
-const complete = (symbol: string, ns: string, callback): void => {
+const complete = (symbol: string, ns: string): Promise<any> => {
     const msg: nREPLCompleteMessage = { op: 'complete', symbol, ns };
-    send(msg).then(respObjs => callback(respObjs[0]));
+    return send(msg).then(respObjs => respObjs[0]);
 };
 
-const info = (symbol: string, ns: string, callback) => {
+const info = (symbol: string, ns: string): Promise<any> => {
     const msg: nREPLInfoMessage = { op: 'info', symbol, ns };
-    send(msg).then(respObjs => callback(respObjs[0]));
+    return send(msg).then(respObjs => respObjs[0]);
 };
 
-const evaluate = (code: string): Promise<any[]> => clone().then((new_session) => {
-    const session_id = new_session['new-session'];
-    const msg: nREPLSingleEvalMessage = { op: 'eval', code: code, session: session_id };
-    return send(msg);
-});
+const evaluate = (code: string): Promise<any[]> => {
+    return clone().then((new_session) => {
+        const session_id = new_session['new-session'];
+        const msg: nREPLSingleEvalMessage = { op: 'eval', code: code, session: session_id };
+        return send(msg);
+    });
+};
 
-const evaluateFile = (code: string, filepath: string): Promise<any[]> => clone().then((new_session) => {
-    const session_id = new_session['new-session'];
-    const msg: nREPLEvalMessage = { op: 'load-file', file: code, 'file-path': filepath, session: session_id };
-    return send(msg);
-});
+const evaluateFile = (code: string, filepath: string): Promise<any[]> => {
+    return clone().then((new_session) => {
+        const session_id = new_session['new-session'];
+        const msg: nREPLEvalMessage = { op: 'load-file', file: code, 'file-path': filepath, session: session_id };
+        return send(msg);
+    });
+};
 
-const stacktrace = (session: string): Promise<any> => send({ op: 'stacktrace', session: session });
+const stacktrace = (session: string): Promise<any> => {
+    return send({ op: 'stacktrace', session: session });
+}
```

Comment:
That's ok. =) It's up to you (and @avli). I'll trust whatever you think it's best then.
If you think the old way was better, just commit those changes back. If not, that's fine as well.